frappe/integrations/doctype/razorpay_settings/razorpay_settings.py

```diff
@@ -317,6 +320,25 @@ def cancel_subscription(self, subscription_id):
 		except Exception:
 			frappe.log_error(frappe.get_traceback())
 
+	def verify_signature(self, body, signature, key):
+		if six.PY3:
+			key = bytes(key, 'utf-8')
+			body = bytes(body, 'utf-8')
+
+		dig = hmac.new(key=key, msg=body, digestmod=hashlib.sha256)
+
+		generated_signature = dig.hexdigest()
+
+		if six.PY2:
```

Comment:
the `develop` branch doesn't support python2 now

Reply:
There are old setups that may upgrade without knowing this, add just for that case. We may need to add a provision in bench to warn them in such cases if it does not exist already?